### PR TITLE
Added FI-JS feature.

### PR DESCRIPTION
**Pull request by srfiorini**
Added FI-JS support to hklib. 

There are also some small fixes do HKGraph.

### CODE REVIEW

File: dist/fi/grammar.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2016-present, IBM Research
+ * Licensed under The MIT License [see LICENSE for details]
+ */
+"use strict";
+// /* TODO */
```
Comment (eltonfss) [inline dist/fi/grammar.js:6]: Is this TODO still needed?

File: src/fi/fi.js [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+    }
+
+    toStringDecorated()
+    {
+        return `<${super.toString()}>`;
+    }
+}
+
+
+//Try to emulate lark idiom for parsing
```
Comment (eltonfss) [inline src/fi/fi.js:251]: What is "lark idiom"

File: src/fi/fi.js [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+    }
+
+}
+
+function processToken(ast)
+{
+    return processJsonPlus(ast.children[0])
+}
+
+function processJsonPlus(ast)
```
Comment (eltonfss) [inline src/fi/fi.js:321]: Is JsonPlus a custom extended Json?

File: src/fi/fi.js [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+    switch (ast.type)
+    {
+        case 'fijs': return processFijs(ast);
+        case 'false': return false;
+        case 'null': return null;
+        case 'true': return true;
+        case 'object': return processObject(ast);
+        case 'array': return processArray(ast);
+        case 'number': return JSON.parse(ast.text);
+        case 'string': return ast.text.substring(1, ast.text.length - 1); //return JSON.parse(ast.text);
```
Comment (eltonfss) [inline src/fi/fi.js:333]: Comment at the end of this line might be removed

File: src/fi/fianchor.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2016-present, IBM Research
+ * Licensed under The MIT License [see LICENSE for details]
+ */
+
+const FIOperator = require("./fioperator");
+
+//const PARSER = new EBNF.Grammars.W3C.Parser(grammar);
```
Comment (eltonfss) [inline src/fi/fianchor.js:8]: Is this comment still needed?

File: src/fi/grammar.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2016-present, IBM Research
+ * Licensed under The MIT License [see LICENSE for details]
+ */
+
+"use strict";
+
+// /* TODO */
```
Comment (eltonfss) [inline src/fi/grammar.js:8]: TODO still needed?

File: src/fi/grammar.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+// boolexpression		::= expression ws* comp_operator ws* expression
+// expression			::= node_att | function | number | boolean | variable | string
+
+module.exports =
+`
+
+fijs				::= id (DOT_OP anchor)*
```
Comment (eltonfss) [inline src/fi/grammar.js:15]: spacing could be standardized

File: src/index.js [hunk 8]
```diff
@@ -67,4 +67,7 @@ exports.HKRepository = require("./datasource/hkrepository");
 
 exports.GraphBuilder = require("./graphbuilder");
 
+exports.FI = require("./fi/fi");
+exports.FIOperator = require("./fi/fioperator")
```
Comment (eltonfss) [inline src/index.js:71]: could add comma at the end of this line

File: src/index.js [hunk 9]
```diff
@@ -67,4 +67,7 @@ exports.HKRepository = require("./datasource/hkrepository");
...
+exports.FIAnchor = require("./fi/fianchor")
```
Comment (eltonfss) [inline src/index.js:72]: this one too

File: test/unit/fi-tests.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+            '<http://example.org/document.txt>.subtext({istart: 2,iend: 5})',
+            '<http://example.org/document.txt>.subtext({istart: 2,iend: 5}).subtext({istart: 0,iend: 2})',
+            '<http://example.org/document.txt>.subtext*({istart: 2,iend: 5})',
+            '``picture1.jpg``.<http://exemple.org/fmask>(``themask.bmp``)',
+            '``picture1.jpg``.<http://exemple.org/fmask>({m: ``themask.bmp``,p: 56})',
+            '``http://www.co-ode.org/ontologies/pizza/pizza.owl#VegetarianPizza``',
+        ];
+
+        for (let strFI of fiToTest){
+            //strFI = strFI.replace(/\s+/g, ''); //remove spaces
```
Comment (eltonfss) [inline test/unit/fi-tests.js:37]: comment still needed?

File: src/fi/fi.js (hunk 3 above)
Comment (srfiorini) [reply src/fi/fi.js:321]: It is not quite json. There is an special treatment of fijs strings (i.e. first case condition).

**Review by eltonfss — APPROVED** [review]

File: test/unit/fi-tests.js (hunk 10 above)
Comment (srfiorini) [reply test/unit/fi-tests.js:37]: Kept as a documentation to anyone trying to deal with this in the future.

**Pull request merged**